generated/nidaqmx/_grpc_time.py

```diff
@@ -30,21 +35,41 @@ def convert_time_to_timestamp(dt: Union[std_datetime, ht_datetime], ts: GrpcTime
         nanos = utc_dt.microsecond * _NS_PER_US
 
     ts.FromNanoseconds(seconds * _NS_PER_S + nanos)
+    return ts
+
+
+def convert_to_desired_timezone(expected_time_utc, tzinfo):
+    current_time_utc = ht_datetime.now(timezone.utc)
+    desired_timezone_offset = current_time_utc.astimezone(tz=tzinfo).utcoffset()
+    desired_expected_time = expected_time_utc + desired_timezone_offset
+    new_datetime = ht_datetime(
+            desired_expected_time.year,
+            desired_expected_time.month,
+            desired_expected_time.day,
+            desired_expected_time.hour,
+            desired_expected_time.minute,
+            desired_expected_time.second,
+            desired_expected_time.microsecond,
+            desired_expected_time.femtosecond,
+            desired_expected_time.yoctosecond,
+            tzinfo = tzinfo)
+    return new_datetime
+
+
+def timestamp_to_1904_epoch(timestamp, yoctoseconds):
+    epoch_1904 = ht_datetime(1904, 1, 1)
+    if timestamp < 0:
+        seconds_from_1904 = ht_timedelta(seconds = _BIAS_FROM_1970_EPOCH - abs(timestamp))
+    else:
+        seconds_from_1904 = ht_timedelta(seconds = _BIAS_FROM_1970_EPOCH + abs(timestamp))
+    datetime_1904 = epoch_1904 + seconds_from_1904 + ht_timedelta(yoctoseconds=yoctoseconds)
```

Comment:
I **think** msb's sign also applies to lsb: for dates before 1904, you should subtract the fractional seconds.

It would be good to find a test case in NI internal time conversion code (e.g. `nitypest`) confirming this.

Reply:
The same logic for the lib time comment applies here